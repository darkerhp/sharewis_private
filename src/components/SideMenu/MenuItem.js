import React, { PropTypes } from 'react';
import ReactNative from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';

const { StyleSheet, Text, View } = ReactNative;

const styles = StyleSheet.create({
  menuWrapper: {
    justifyContent: 'flex-start',
    alignItems: 'center',
    flexDirection: 'row',
    marginBottom: 30,
  },
  menuIcon: {
    fontSize: 32,
    color: '#999',
    marginRight: 10,
  },
  menuText: {
    fontSize: 17,
    color: '#33bcd0',
  },
});

const MenuItem = ({ text, iconName, handlePress }) =>
  (
    <View style={styles.menuWrapper}>
      <Icon style={styles.menuIcon} name={iconName} />
      <Text style={styles.menuText} onPress={handlePress}>
        {text}
      </Text>
    </View>
  );

MenuItem.propTypes = {
  text: PropTypes.string.isRequired,
  iconName: PropTypes.string.isRequired,
  handlePress: PropTypes.func.isRequired,
};

export default MenuItem;
