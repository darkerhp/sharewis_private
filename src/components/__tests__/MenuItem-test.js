/* eslint-disable no-undef */
/* eslint-disable import/no-extraneous-dependencies */
import 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import MenuItem from '../SideMenu/MenuItem';

describe('MenuItem', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MenuItem text={'text'} iconName={'mail'} handlePress={() => null} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
