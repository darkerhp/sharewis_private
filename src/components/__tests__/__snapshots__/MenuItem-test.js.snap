exports[`MenuItem renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "marginBottom": 30,
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Object {
          "color": "#999",
          "fontSize": 32,
          "marginRight": 10,
        },
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }>
    î…˜
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "color": "#33bcd0",
        "fontSize": 17,
      }
    }>
    text
  </Text>
</View>
`;
